figtree.go

```
@@ -576,8 +719,23 @@ func (m *Merger) makeMergeStruct(values ...reflect.Value) reflect.Value {
 	}
 
 	fields := []reflect.StructField{}
+	seen := map[string]reflect.StructField{}
+	yamlSeen := map[string]reflect.StructField{}
 	for _, value := range foundFields {
 		fields = append(fields, value)
+		if prev, ok := seen[value.Name]; ok {
+			// we have a duplicate field name, this should not happen
+			// but if it does, we will just skip it
+			fmt.Fprintf(os.Stderr, "Duplicate field name %q in merge struct.\n\tOld: %s %s `%s`\n\tNew: %s %s `%s`\n", value.Name, prev.Name, prev.Type.String(), prev.Tag, value.Name, value.Type.String(), value.Tag)
```

Is this debugging code? Not sure we want figtree to print to `os.Stderr` on its own accord. I'd suggest either:

* A panic if this shouldn't be possible to trigger
* Silently ignored if it can happen but isn't actionable
* Pass in a logger if it's something we want to bubble up non-fatally

---

I ended up converting these to panics.  The reflect.MakeStruct call will panic below if we have duplicate fields, so better to panic here with slightly friendlier output.